Issue installation process

In a deployment on my own computer, the button "Continue" on the page shown in the screenshot below works fine, and the second step is shown afterwards.

However, in the deployment on an internal server, the button "Continue" on the first page results in an error in the log, and in the first page being shown again.

![image](https://user-images.githubusercontent.com/11427611/138945533-4cbcc6be-6491-42ff-a147-b89e1b105dc6.png)

I added some debugging (see https://github.com/heidivanparys/re3gistry/commits/additional_changes_2 for the code used for the application).

On my computer

- Windows 10
- Java AdoptOpenJDK x64_jdk8u265-b01
- Apache Tomcat 8.5.70
- Apache Solr 8.9.0
- no Apache Server

when calling re3gistry2/install I get:

```
[DEBUG] 2021-10-26 20:57:27.518 [http-nio-8088-exec-1] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 20:57:27.545 [http-nio-8088-exec-1] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[TRACE] 2021-10-26 20:57:27.590 [http-nio-8088-exec-1] [controller.Install.processRequest:53] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-10-26 20:57:27.591 [http-nio-8088-exec-1] [utility.Configuration.initServlet:166] - Start initialising servlet
[DEBUG] 2021-10-26 20:57:27.595 [http-nio-8088-exec-1] [utility.Configuration.initServlet:184] - Proceed initialising servlet
[TRACE] 2021-10-26 20:57:27.595 [http-nio-8088-exec-1] [utility.Configuration.initServlet:187] - Initialise properties
[TRACE] 2021-10-26 20:57:27.595 [http-nio-8088-exec-1] [utility.Configuration.initServlet:192] - Initialise localisation
[TRACE] 2021-10-26 20:57:27.596 [http-nio-8088-exec-1] [utility.Configuration.initServlet:197] - Initialise available languages in session if needed
[DEBUG] 2021-10-26 20:57:27.597 [http-nio-8088-exec-1] [utility.Configuration.initServlet:209] - Current page name=/install
[DEBUG] 2021-10-26 20:57:27.597 [http-nio-8088-exec-1] [utility.Configuration.initServlet:217] - Check user
[TRACE] 2021-10-26 20:57:35.604 [http-nio-8088-exec-1] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-10-26 20:57:35.612 [http-nio-8088-exec-1] [utility.Configuration.initServlet:225] - No user found
[DEBUG] 2021-10-26 20:57:35.655 [http-nio-8088-exec-1] [controller.Install.processRequest:72] - install step = 1
[DEBUG] 2021-10-26 20:58:09.414 [http-nio-8088-exec-5] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 20:58:09.415 [http-nio-8088-exec-5] [controller.ValidateSalt.doFilter:65] - Salt is in cache, thus moving on
[TRACE] 2021-10-26 20:58:09.416 [http-nio-8088-exec-5] [controller.Install.processRequest:53] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-10-26 20:58:09.416 [http-nio-8088-exec-5] [utility.Configuration.initServlet:166] - Start initialising servlet
[DEBUG] 2021-10-26 20:58:09.417 [http-nio-8088-exec-5] [utility.Configuration.initServlet:184] - Proceed initialising servlet
[TRACE] 2021-10-26 20:58:09.418 [http-nio-8088-exec-5] [utility.Configuration.initServlet:187] - Initialise properties
[TRACE] 2021-10-26 20:58:09.418 [http-nio-8088-exec-5] [utility.Configuration.initServlet:192] - Initialise localisation
[TRACE] 2021-10-26 20:58:09.418 [http-nio-8088-exec-5] [utility.Configuration.initServlet:197] - Initialise available languages in session if needed
[DEBUG] 2021-10-26 20:58:09.418 [http-nio-8088-exec-5] [utility.Configuration.initServlet:209] - Current page name=/install
[DEBUG] 2021-10-26 20:58:09.418 [http-nio-8088-exec-5] [utility.Configuration.initServlet:217] - Check user
[TRACE] 2021-10-26 20:58:09.583 [http-nio-8088-exec-5] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-10-26 20:58:09.597 [http-nio-8088-exec-5] [utility.Configuration.initServlet:225] - No user found
[DEBUG] 2021-10-26 20:58:10.118 [http-nio-8088-exec-5] [controller.Install.processRequest:72] - install step = 2
```

On the server

- Ubuntu v20.04LTS
- Java 1.8.0_292-8u292-b10-0ubuntu1~20.04-b10
- Apache Tomcat 8.5.70
- Apache Solr 8.9.0
- Apache Server installed, but the application on the Tomcat is accessed directly

```
[DEBUG] 2021-10-26 21:04:04.926 [http-nio-8080-exec-26] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:04.932 [http-nio-8080-exec-26] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[DEBUG] 2021-10-26 21:04:05.208 [http-nio-8080-exec-27] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:05.209 [http-nio-8080-exec-27] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[TRACE] 2021-10-26 21:04:05.210 [http-nio-8080-exec-27] [controller.Login.processRequest:42] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Login
[DEBUG] 2021-10-26 21:04:05.210 [http-nio-8080-exec-27] [utility.Configuration.initServlet:162] - Start initialising servlet
[TRACE] 2021-10-26 21:04:05.211 [http-nio-8080-exec-27] [utility.Configuration.initServlet:177] - Redirecting to ./install
[ERROR] 2021-10-26 21:04:05.211 [http-nio-8080-exec-27] [controller.Login.doGet:64] - Cannot forward after response has been committed
java.lang.IllegalStateException: Cannot forward after response has been committed
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:326) ~[catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:316) ~[catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.Login.processRequest(Login.java:54) ~[classes/:?]
	at eu.europa.ec.re3gistry2.web.controller.Login.doGet(Login.java:61) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:655) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:47) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:61) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:387) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.ValidateSalt.doFilter(ValidateSalt.java:78) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.LoadSalt.doFilter(LoadSalt.java:72) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.14.1.jar:2.14.1]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:196) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:8.5.70]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:544) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:8.5.70]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:364) [catalina.jar:8.5.70]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:624) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:831) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1650) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.70]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
[DEBUG] 2021-10-26 21:04:05.417 [http-nio-8080-exec-28] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:05.418 [http-nio-8080-exec-28] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[TRACE] 2021-10-26 21:04:05.418 [http-nio-8080-exec-28] [controller.Install.processRequest:53] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-10-26 21:04:05.419 [http-nio-8080-exec-28] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-10-26 21:04:05.419 [http-nio-8080-exec-28] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-10-26 21:04:05.419 [http-nio-8080-exec-28] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-10-26 21:04:05.419 [http-nio-8080-exec-28] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-10-26 21:04:05.420 [http-nio-8080-exec-28] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-10-26 21:04:05.421 [http-nio-8080-exec-28] [utility.Configuration.initServlet:205] - Current page name=/install
[DEBUG] 2021-10-26 21:04:05.421 [http-nio-8080-exec-28] [utility.Configuration.initServlet:213] - Check user
[TRACE] 2021-10-26 21:04:07.084 [http-nio-8080-exec-28] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-10-26 21:04:07.087 [http-nio-8080-exec-28] [utility.Configuration.initServlet:221] - No user found
[DEBUG] 2021-10-26 21:04:07.099 [http-nio-8080-exec-28] [controller.Install.processRequest:72] - install step = 1
[DEBUG] 2021-10-26 21:04:28.269 [http-nio-8080-exec-31] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:28.269 [http-nio-8080-exec-31] [controller.ValidateSalt.doFilter:70] - Salt is not in cache, and HTTP method is  POST, thus potential CSRF
[ERROR] 2021-10-26 21:04:28.270 [http-nio-8080-exec-31] [controller.ValidateSalt.doFilter:74] - Potential CSRF detected
[DEBUG] 2021-10-26 21:04:28.368 [http-nio-8080-exec-32] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:28.368 [http-nio-8080-exec-32] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[TRACE] 2021-10-26 21:04:28.369 [http-nio-8080-exec-32] [controller.Login.processRequest:42] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Login
[DEBUG] 2021-10-26 21:04:28.369 [http-nio-8080-exec-32] [utility.Configuration.initServlet:162] - Start initialising servlet
[TRACE] 2021-10-26 21:04:28.370 [http-nio-8080-exec-32] [utility.Configuration.initServlet:177] - Redirecting to ./install
[ERROR] 2021-10-26 21:04:28.370 [http-nio-8080-exec-32] [controller.Login.doGet:64] - Cannot forward after response has been committed
java.lang.IllegalStateException: Cannot forward after response has been committed
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:326) ~[catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:316) ~[catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.Login.processRequest(Login.java:54) ~[classes/:?]
	at eu.europa.ec.re3gistry2.web.controller.Login.doGet(Login.java:61) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:655) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:47) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:61) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:387) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.ValidateSalt.doFilter(ValidateSalt.java:78) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.LoadSalt.doFilter(LoadSalt.java:72) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.14.1.jar:2.14.1]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:196) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:8.5.70]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:544) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:8.5.70]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:364) [catalina.jar:8.5.70]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:624) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:831) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1650) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.70]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
[DEBUG] 2021-10-26 21:04:28.420 [http-nio-8080-exec-18] [controller.ValidateSalt.doFilter:57] - Validating that the salt is in the cache
[DEBUG] 2021-10-26 21:04:28.420 [http-nio-8080-exec-18] [controller.ValidateSalt.doFilter:77] - Salt is not in cache, but HTTP method is not POST, thus moving on
[TRACE] 2021-10-26 21:04:28.421 [http-nio-8080-exec-18] [controller.Install.processRequest:53] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-10-26 21:04:28.421 [http-nio-8080-exec-18] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-10-26 21:04:28.421 [http-nio-8080-exec-18] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-10-26 21:04:28.422 [http-nio-8080-exec-18] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-10-26 21:04:28.422 [http-nio-8080-exec-18] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-10-26 21:04:28.422 [http-nio-8080-exec-18] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-10-26 21:04:28.423 [http-nio-8080-exec-18] [utility.Configuration.initServlet:205] - Current page name=/install
[DEBUG] 2021-10-26 21:04:28.423 [http-nio-8080-exec-18] [utility.Configuration.initServlet:213] - Check user
[TRACE] 2021-10-26 21:04:28.448 [http-nio-8080-exec-18] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-10-26 21:04:28.450 [http-nio-8080-exec-18] [utility.Configuration.initServlet:221] - No user found
[DEBUG] 2021-10-26 21:04:28.463 [http-nio-8080-exec-18] [controller.Install.processRequest:72] - install step = 1
```

So on the server

- `eu.europa.ec.re3gistry2.web.controller.Login` is called
- `java.lang.IllegalStateException: Cannot forward after response has been committed` is thrown
- there is a message `Potential CSRF detected`

Any help with identifying the problem behind this would be very welcome.

--- a/documentation/administrator-manual.md
+++ b/documentation/administrator-manual.md
@@ -92,7 +92,7 @@ The last step is to start tomcat.
 
 ### Installation wizard
 
-To start the installation wizard, visit the following address: `http://<tomcat-url>:<tomcat-port>/re3gistry2/install` (e.g. `http://localhost:8080/re3gistry2/install`).
+To start the installation wizard, visit the following address: `http://<tomcat-url>:<tomcat-port>/re3gistry2/install` (e.g. `http://localhost:8080/re3gistry2/install`). Note: do not visit `http://<tomcat-url>:<tomcat-port>/re3gistry2`, as the welcome page of the application is `http://localhost:8080/re3gistry2/login` (see sources/Re3gistry2/src/main/webapp/WEB-INF/web.xml), and that page should only be visited after a successful installation.
 
 The first page is just an introductory description. Click on "continue".
 
@@ -105,7 +105,7 @@ The next step allows you to choose between 2 options:
 *  Installation: choose this option if you want to install the Re3gistry software for the first time and to create a registry from scratch
 * Migration: choose this option if you want to upgrade a preexisting Re3gistry instance. This option will keep your settings and content. Note that this option **will work only on an existing instance running on Re3gistry version 1.3.X**.
 
-#### Installaion option
+#### Installation option
 
 In case the "Installation" option is chosen, the basic configuration for the new registry will be requested.
 

--- a/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/LoadSalt.java
+++ b/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/LoadSalt.java
@@ -26,6 +26,8 @@ package eu.europa.ec.re3gistry2.web.controller;
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 import eu.europa.ec.re3gistry2.base.utility.BaseConstants;
+import eu.europa.ec.re3gistry2.base.utility.Configuration;
+
 import java.io.IOException;
 import java.security.SecureRandom;
 import java.util.concurrent.TimeUnit;
@@ -47,12 +49,15 @@ public class LoadSalt implements Filter {
 
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        Configuration.getInstance().getLogger().trace("Start filter " + ValidateSalt.class.getName());
+      
         HttpServletRequest httpReq = (HttpServletRequest) request;
 
         // Check the user session for the salt cache, if none is present we create one
         Cache<String, Boolean> csrfPreventionSaltCache = (Cache<String, Boolean>) httpReq.getSession().getAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALTCACHE);
 
         if (csrfPreventionSaltCache == null) {
+            Configuration.getInstance().getLogger().trace("Creating CSRF prevention salt cache");
             csrfPreventionSaltCache = CacheBuilder.newBuilder()
                     .maximumSize(5000)
                     .expireAfterWrite(20, TimeUnit.MINUTES)
@@ -61,6 +66,7 @@ public class LoadSalt implements Filter {
             httpReq.getSession().setAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALTCACHE, csrfPreventionSaltCache);
         }
 
+        Configuration.getInstance().getLogger().trace("Generate the salt, store it in the users cache, and add it to the current request");
         // Generate the salt and store it in the users cache
         String salt = RandomStringUtils.random(20, 0, 0, true, true, null, new SecureRandom());
         csrfPreventionSaltCache.put(salt, Boolean.TRUE);

--- a/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/ValidateSalt.java
+++ b/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/ValidateSalt.java
@@ -47,32 +47,35 @@ public class ValidateSalt implements Filter {
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
         throws IOException, ServletException {
+        Configuration.getInstance().getLogger().trace("Start filter " + ValidateSalt.class.getName());
 
         // Assume its HTTP
         HttpServletRequest httpReq = (HttpServletRequest) request;
 
         // Get the salt sent with the request
-        String salt = (String) httpReq.getParameter(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALT);
+        String salt = (String) httpReq.getAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALT);
 
-        // Validate that the salt is in the cache
+        Configuration.getInstance().getLogger().debug("Validating that the salt is in the cache");
         Cache<String, Boolean> csrfPreventionSaltCache = (Cache<String, Boolean>)
-            httpReq.getSession().getAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALTCACHE);
+        httpReq.getSession().getAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTIONSALTCACHE);
 
         if (csrfPreventionSaltCache != null &&
                 salt != null &&
                 csrfPreventionSaltCache.getIfPresent(salt) != null){
 
-            // If the salt is in the cache, we move on
+            Configuration.getInstance().getLogger().debug("Salt is in cache, thus moving on");
             chain.doFilter(request, response);
         } else {
             // Otherwise we throw an exception aborting the request flow
             if(httpReq.getMethod().equals("POST")){
+                Configuration.getInstance().getLogger().debug("Salt is not in cache, and HTTP method is POST, thus potential CSRF");
                 request.setAttribute(BaseConstants.KEY_REQUEST_CSRF_PREVENTION_ERROR, BaseConstants.KEY_BOOLEAN_STRING_TRUE);
                 request.getRequestDispatcher(WebConstants.PAGE_JSP_FOLDER + WebConstants.PAGE_PATH_INDEX + WebConstants.PAGE_URINAME_INDEX + WebConstants.PAGE_JSP_EXTENSION).forward(request, response);
                 Logger logger = Configuration.getInstance().getLogger();
                 logger.error("Potential CSRF detected");
                 //throw new ServletException("Potential CSRF detected");
             }else{
+                Configuration.getInstance().getLogger().debug("Salt is not in cache, but HTTP method is not POST, thus moving on");
                 chain.doFilter(request, response);
             }
         }

